baseplate/thrift/baseplate.thrift

```diff
@@ -127,3 +127,161 @@ struct Request {
     5: OriginService origin_service;
     6: Geolocation geolocation;
 }
+
+/** The integer values within this enum correspond to HTTP status codes.
+
+HTTP layers can easily map errors to an appropriate status code.
+*/
+enum ErrorCode {
+    /** This is appropriate if the client sent an invalid request.
+
+    You can send the details using the details map in the Error struct.
+    */
+    BAD_REQUEST = 400,
+    /** This is appropriate when you fail to authenticate the request.
+
+    It may be appropriate for the client to retry this request in the event, for
+    example, if they used an expired authentication credential, they can retry
+    */
+    UNAUTHORIZED = 401,
+    /** This is appropriate if you need to communicate to the client that their
+    request can not be completed until a payment is made.
+    */
+    PAYMENT_REQUIRED = 402,
+    /** This is appropriate when you can authenticate a request but the client
+    does not have access to the requested resource.
+
+    Unlike Unauthorized, refreshing an authentication resource and trying again
+    will not make a difference.
+    */
+    FORBIDDEN = 403,
+    /** This is appropriate when the client tries to access something that does
+    not exist.
+    */
+    NOT_FOUND = 404,
+    /** This is appropriate when a client request would cause a conflict with
```

Comment:
some of these are framed like "when the client does X" while others are "a client request does X". It'd be nice to have a consistent form.